src/components/reviews/SingleReview.jsx

```diff
@@ -1,71 +1,210 @@
+/* eslint-disable max-len */
 import React, { useState } from 'react';
+import styled from 'styled-components';
+import { format, parseISO } from 'date-fns';
+import { GrCheckmark } from 'react-icons/gr';
+import Stars from '../shared/Stars.jsx';
+import { markHelpful, reportReview } from './api.js';
+import useModal from '../shared/useModal.js';
+import Modal from '../shared/Modal.jsx';
+
+const Review = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const UserDate = styled.div`
+  margin-left: auto;
+`;
+
+const ReviewSummary = styled.div`
+  font-weight: bold;
+`;
+
+const ReviewSummaryOverflow = styled.div``;
+
+const ReviewBody = styled.div``;
+
+const ShowMoreButton = styled.div``;
+
+const ImageContainer = styled.div`
+  display: flex;
+`;
+
+const SingleImage = styled.div``;
+
+const Image = styled.img`
+  width: 45px;
+  height: auto;
+  padding: 10px;
+`;
+
+const ModalContainer = styled.div``;
+
+const ModalImage = styled.img``;
+
+const Recommend = styled.div``;
+
+const ResponseContainer = styled.div``;
+
+const ResponseHeader = styled.div`
+  font-weight: bold;
+`;
+
+const ResponseBody = styled.div``;
+
+const HelpfulReportContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const HelpfulReportButton = styled.div`
+  text-decoration: underline;
+`;
+
+const Line = styled.hr`
+  display: block;
+  height: 1px;
+  border: 0;
+  border-top: 1px solid white;
+  margin: 1em 0;
+  padding: 0;
+`;
 
 function SingleReview(props) {
   const { review } = props;
-
-  const [showMore, setShowMore] = useState(false);
   // this is my example review - leaving it in for future use
   // const review = {
   //   review_id: 1277210,
   //   rating: 3,
-  //   summary: 'Such a great product!',
+  //   summary: 'Such a great product! sdfadsfklajdsklfjdlkajsfldsafsafdsfdsdsfsdfdsfdssd',
   //   recommend: true,
-  //   response: null,
-  //   body: 'wow I really loved this product. It was perfect',
+  //   response: 'dfhkjdahskfhdaskjhjk',
+  //   body: 'wow I really loved this product. It was perfect wow I really loved this product. It was perfect wow I really loved this product. It was perfect wow I really loved this product. It was perfectwow I really loved this product. It was perfectwow I really loved this product. It was perfect',
   //   date: '2022-10-25T00:00:00.000Z',
   //   reviewer_name: 'cordelia',
   //   helpfulness: 2,
-  //   photos: [],
+  //   photos: ['https://candicehern.com/WP/wp-content/uploads/2013/04/evening_dress3.jpg', 'https://images.fineartamerica.com/images/artworkimages/mediumlarge/2/carriage--ball-dresses-print-collector.jpg', 'https://thegraphicsfairy.com/wp-content/uploads/2020/07/Edwardian-Fashion-Image-GraphicsFairy.jpg'],
   // };
+  const [showMore, setShowMore] = useState(false);
+  const [helpful, setHelpful] = useState(review.helpfulness);
+  const { visible, toggle } = useModal();
```

Comment:
I think this is where the issue with the modal might coming from. When one image gets clicked, `visible` gets set to true for the whole `SingleReview` component, which means it's also true for every `Modal` instance that gets made when you map over `review.photos`. This causes every image modal to pop up at the same time. I think a possible fix is to separate `ImageContainer` into its own file, and then call `useModal` within that component instead of in `SingleReview`. That way each image has its own `visible` variable.

Reply:
that was definitely the issue! I made a separate image filed and that solved the problem